packages/playwright/src/worker/testInfo.ts

```diff
@@ -454,6 +454,7 @@ export class TestInfoImpl implements TestInfo {
         .replace(/\{(.)?projectName\}/g, projectNamePathSegment ? '$1' + projectNamePathSegment : '')
         .replace(/\{(.)?testName\}/g, '$1' + this._fsSanitizedTestName())
         .replace(/\{(.)?testFileName\}/g, '$1' + parsedRelativeTestFilePath.base)
+        .replace(/\{(.)?testFileNameWithoutExtension\}/g, '$1' + parsedRelativeTestFilePath.base.replace('.spec', '').replace('.ts', ''))
```

Comment:
There is also at least *.spec.js, *.spec.tsx, *.spec.jsx, and same with `.test.` in the name.